### PR TITLE
chore: Remove old way of flagging features as experimental

### PR DESCRIPTION
**Pull request by adchia**
**What this PR does / why we need it**:
For a while now, we've been using warnings as the default way to tell users a feature is experimental. This formally deprecates needing `feature_store.yaml` flags to enable ODFV and lambda feature servers.

**Which issue(s) this PR fixes**:

Fixes #

### CODE REVIEW

File: sdk/python/docs/source/feast.rst [hunk 1]
```diff
@@ -169,14 +169,6 @@ feast.field module
    :undoc-members:
    :show-inheritance:
 
-feast.flags module
```
Comment (felixwang9817) [inline sdk/python/docs/source/feast.rst:172]: does `sdk/python/docs/index.rst` also need to be updated? not sure which of these two is the source of truth

File: docs/reference/feature-servers/alpha-aws-lambda-feature-server.md [hunk 2]
```diff
@@ -41,12 +34,21 @@ If enabled, the feature server will be deployed during `feast apply`. After it i
 
 Feast requires the following permissions in order to deploy and teardown AWS Lambda feature server:
 
+<<<<<<< HEAD:docs/reference/feature-servers/alpha-aws-lambda-feature-server.md
```
Comment (felixwang9817) [inline docs/reference/feature-servers/alpha-aws-lambda-feature-server.md:37]: rebase issues here

Comment (adchia) [reply docs/reference/feature-servers/alpha-aws-lambda-feature-server.md]: done

File: sdk/python/docs/source/feast.rst (hunk 1 above)
Comment (adchia) [reply sdk/python/docs/source/feast.rst:172]: nope. They are just different. index.rst gives the top level structure and this isn't referenced there.

This change was generated by running `make build-sphinx`

**Review by felixwang9817 — APPROVED** [review]
/lgtm

**Pull request merged**